fabtests/benchmarks/benchmark_shared.c

```diff
@@ -155,9 +225,42 @@ int pingpong(void)
 	if (ret)
 		return ret;
 
-	ret = pingpong_pre_posted_rx(inject_size);
-	if (ret)
-		return ret;
+	if (ft_check_opts(FT_OPT_NO_PRE_POSTED_RX)) {
+		/* Consume pre-posted RX buffers on client and server. */
+		if (opts.dst_addr) {
+			ret = ft_tx(ep, remote_fi_addr, opts.transfer_size,
+				    &tx_ctx);
+			if (ret)
+				return ret;
+
+			ret = ft_get_rx_comp(rx_seq);
+			if (ret)
+				return ret;
+		} else {
+			ret = ft_get_rx_comp(rx_seq);
+			if (ret)
+				return ret;
+
+
+			ret = ft_tx(ep, remote_fi_addr, opts.transfer_size,
+				    &tx_ctx);
+			if (ret)
+				return ret;
+		}
```

Comment:
Can you move this into the init_fabric flow so that other tests can use it as well?

Reply:
I can. What this logic is is an inband sync without RX repost. I'll make a function for this which can be conditionally called in ft_init_fabric.